### DISCUSSION TITLE
VGA experiments

### DISCUSSION THREAD

**Discussion by bwhitman** [shorepine/tulipcc#17] [General]
I took my Tulip breadboard edition (tm) and ran the R7, G7, B7, HSYNC, VSYNC and GND cables to a VGA port. At 640x480 I do eventually get a sync (at 53Hz, under spec) and see the first signs of life of a VGA output Tulip. The next ESP32-LCD capable display clock up, 28MHz (from 22MHz) does not sync at all, claiming an FPS of 80Hz.  I'll try to find a resolution / clock pair that stays closer to 60Hz and debug from there. 

![IMG_0544](https://user-images.githubusercontent.com/76612/221963621-f8325309-2106-47d4-b8ea-b5d4819aa09e.jpeg)

![IMG_0542](https://user-images.githubusercontent.com/76612/221963658-fbf8cf90-325b-484b-a201-3cfd85a775ee.jpeg)

**Comment by gramster** [comment]
Maybe related to this, it would be great if Tulip supported external monitors over USB-C. I guess its not practical, but my eyes are not that great at dealing with tiny text anymore.

**Reply by bwhitman** [reply]
Digital video out (that's not TTLRGB like we use now) would be very very hard without an external chip / FPGA to do the buffering. The ESP32S3 has special fast GDMA output that lets us write to RGBTTL (and maybe VGA one day, would love to pick this back up) but it doesn't let you write arbitrary packets like a DVI / USBC setup would want. 

May be easier to run Tulip Desktop on a RPi for now on a bigger screen.
